### DISCUSSION TITLE
Collapse duplicate results

### DISCUSSION THREAD

**Discussion by nshp** [community/community#9831] [Other Feature Feedback, Questions, & Ideas]
One of the most frustrating things about the existing GitHub search for me is how I often find page after page of almost exactly the same result, and this still appears to be true of the new code search.

Example: https://github.com/search?q=0x8001180000000d38&type=code
vs new: https://cs.github.com/?scopeName=All+repos&scope=&q=0x8001180000000d38

This happens most with results from the Linux kernel, embedded board support packages, etc. which are duplicated across many repos without being proper "forks". Sometimes the whole file isn't the same, but the entire snippet in the search results is. Sometimes there are non-duplicate results buried on page 13 out of 53. Can I tell? No.

These results shouldn't be _inaccessible_, but I'd want to have to click through a "See [X] similar results" to see them.

**Comment by grishavanika** [comment]
Yeap, option to "show best match per repo/forks" is my dream.
